colossalai/engine/schedule/_pipeline_schedule.py

```diff
@@ -386,8 +387,7 @@ def __init__(self,
         self.num_model_chunks = num_model_chunks
 
     def pre_processing(self, engine):
-        # FIXME(jiaruifang) we shall not use ShardedModelV2 in pipeline mode, due to circular dependency.
-        if hasattr(engine.model, 'colo_attr'):
+        if isinstance(engine.model[0], ShardedModelV2):
```

Comment:
Maybe we can use `isinstance(engine.model[0], (ShardedModelV2, NaiveAMPModel))`

Reply:
This is a bug. It should be if isinstance(engine.model, ShardedModelV2):
I fixed it